### ISSUE TITLE
TreeView `isChecked` not working as expected

### ISSUE THREAD

**Issue by RoshanSharma1** [telerik/kendo-angular#1535]
### Bug report

[https://plnkr.co/edit/7iQq6jylHlKGZvfejkn3?p=preview](https://plnkr.co/edit/7iQq6jylHlKGZvfejkn3?p=preview)

### Reproduction of the problem

The state of the checkbox is not controlled by the `isChecked` functionality. According to documentation, checkbox should always be unchecked in above example.

### Current behavior
Initially, on first load, the behavior is working properly but after clicking on any of the checkboxes, the checkboxes show as checked.

### Expected/desired behavior
The checkboxes should be displayed as per the `isChecked` function.

**Comment by ggkrustev** [comment]
Hi @RoshanSharma1, thanks for the report.

Indeed the TreeView leaves the clicked checkbox checked, because the there is no dedicated change detection cycle for its resetting. We will look into the case and will update the thread once we have a working solution.

**Comment by RoshanSharma1** [comment]
@ggkrustev, I am waiting for this issue. Is there any update for this issue ? When will be this resolved ?

**Comment by ggkrustev** [comment]
I am afraid that there is no specific ETA for the time being. We will schedule it for further investigation once there are tasks with less priority.

For now, the best option is to customize the built-in CheckDirective in order to achieve your specific goal. You can use our custom directive as a base line.

https://www.telerik.com/kendo-angular-ui/components/treeview/checkboxes/#toc-custom

We will update the thread once we have a working DEV release.

**Comment by heinzer** [comment]
Is there any update on this bug? I've tried the custom check directive, but linked above, but I'm also running across issues even in the plunker. 

I modified the custom checkbox plunker to exclude a certain tree node conditionally (if it has an index of '0'). https://plnkr.co/edit/uxXis6DI7wVLR4b2FtD4?p=info

If I want to not check the item at index '0', it will still check the checkbox. Additionally, since this is not limited by the checked keys now, i can run into a scenario where two checkboxes will be checked when there should only be one (the item at index '0' as well as any other checkbox).
